### DISCUSSION TITLE
helmv3: exclude stable repository

### DISCUSSION THREAD

**Discussion by jgarec** [renovatebot/renovate#27299] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.170.0 (renovate-ce)

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

Hello,

We are behind a proxy and we have our nexus mirroring most of the public repositories. 
I want to avoid to run the command "helm repo add stable https://charts.helm.sh/stable --force-update" because we don't use the stable repository and the command won't work (`Error looks like \"https://charts.helm.sh/stable\" is not a valid chart repository or cannot be reached...`)
I just want to use my own repo so my config looks like : 
```
helmv3: {
  registryAliases: {
    myrepo: "https://myrepo/"
  }
}
```

but if i understand well, since #26848 it will merge the default config (https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/helmv3/index.ts#L10-L16) with mine ?

Is there a way to globally exclude this default repository ?

Thx for your help.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
We don't really have a way to _exclude_ registries but you can set the default registry for a datasource, or manually override to one or more registries using packageRules + registryUrls

**Comment by jgarec** [comment]
humm i still don't find a clear way to override the default registryAlias whole object.

The only solution i've found is to update the value of stable registryAlias to the private endpoint we use : 
```
helmv3: {
  registryAliases: {
    stable: "https://myrepo/",
    myrepo: "https://myrepo/"
  }
}
```

**Comment by svifred** [comment]
@jgarec I've used and amended your approach to "null out" the Helm stable repo:
```
"helmv3": {
  "registryAliases": {"stable": ""}
}
```
This inhibits `helm repo add stable`. Thanks for the pointer!

@jamietanna Before trying the above, which does feel like a hack or brittle workaround, I tried defining a `hostRule` disabling `charts.helm.sh`. Much to my surprise, this had no effect.

To be clear, when adding
```
hostRules: [
  {
    "matchHost": "charts.helm.sh",
    "enabled": false
  }
]
```
to `config.js`, I expect all requests to `charts.helm.sh` to be skipped or filtered out, but logs clearly shows `helm repo add stable https://charts.helm.sh/stable --force-update` (which then fails due to a proxy setup denying requests to that host).

Is this a bug? Or is it expected, but undocumented? Or have I failed to find the proper docs on handling hosts for Helm dependencies?
